### PR TITLE
Improve PSBT explanation

### PR DESCRIPTION
**Pull request by dennisreimann**
Inspired by https://github.com/bitcoin/bitcoin/blob/master/doc/psbt.md#psbt-in-general

Gets linked to in btcpayserver/btcpayserver#2559

### CODE REVIEW

**Review by Zaxounette — APPROVED** [review]
LGTM.

I'll probably also add a fully fledged non-technical paragraph at the bottom of this page in the already present #PSBT section.

**Comment by dennisreimann** [comment]
Cool, you can go ahead an directly add it to this branch -- I just granted you access to my fork.

**Comment by Zaxounette** [comment]
And I accepted the invitation, My Lord.

**Comment by dennisreimann** [comment]
The main PR got merged, so we should merge this one before the next release too.

File: docs/Wallet.md [hunk 1]
```diff
@@ -50,9 +50,9 @@ The Send function allows **spending of the funds from the BTCPay wallet**.
 
 To spend the funds, you are required to **sign** the transaction. Transactions can be signed with:
 
-- HD Private key or mnemonic seed
-- Wallet supporting PSBT
 - Hardware Wallet
+- Wallet supporting PSBT
+- Private key or recovery seed
```
Comment (Kukks) [inline docs/Wallet.md:55]: It's still an HD private key though (xpriv)

**Review by Kukks — APPROVED** [review]

File: docs/Wallet.md (hunk 1 above)
Comment (dennisreimann) [inline docs/Wallet.md:55]:
```suggestion
- HD private key or recovery seed
```

**Comment by dennisreimann** [comment]
@Zaxounette I'll merge this one and you can open a separate PR for your additions.

**Pull request merged**